tests/rbd_mirror/test_rbd_group_consistency.py

```diff
@@ -0,0 +1,407 @@
+    primary_cluster,
+    secondary_cluster,
+    pool_types,
+    **kw,
+):
+    """
+    Test group level data consistency on secondary after primary site disaster
+    Args:
+        rbd_primary: RBD object of primary cluster
+        rbd_secondary: RBD objevct of secondary cluster
+        client_primary: client node object of primary cluster
+        client_secondary: client node object of secondary cluster
+        primary_cluster: Primary cluster object
+        secondary_cluster: Secondary cluster object
+        pool_types: Replication pool or EC pool
+        **kw: any other arguments
+    """
+
+    for pool_type in pool_types:
+        rbd_config = kw.get("config", {}).get(pool_type, {})
+        multi_pool_config = deepcopy(getdict(rbd_config))
+        log.info("Running test CEPH-83613291  for %s", pool_type)
+
+        for pool, pool_config in multi_pool_config.items():
+            group_config = {}
+            if "data_pool" in pool_config.keys():
+                _ = pool_config.pop("data_pool")
+            group_spec = pool_config.get("group-spec")
+            group_config.update({"group-spec": group_spec})
+
+            image_spec_small = []
+            image_spec_large = []
+            for image, image_config in pool_config.items():
+                if "image" in image:
+                    if "namespace" in pool_config:
+                        pool_spec = pool + "/" + pool_config.get("namespace") + "/"
+                    else:
+                        pool_spec = pool + "/"
+                    image_spec_small.append(pool_spec + image)
+                    image_large_name = pool_spec + "image_" + random_string(len=4)
+                    # Create large size images
+                    rbd_primary.create(
+                        **{"image-spec": image_large_name, "size": "10G"}
+                    )
+                    image_spec_large.append(image_large_name)
+                    if add_image_to_group_and_verify(
+                        **{
+                            "group-spec": group_spec,
+                            "image-spec": image_large_name,
+                            "client": client_primary,
+                        }
+                    ):
+                        raise Exception("Failed to add image to group %s", group_spec)
+
+            if "namespace" in pool_config:
+                enable_namespace_mirroring(
+                    rbd_primary, rbd_secondary, pool, **pool_config
+                )
+
+            image_spec_all = []
+            image_spec_all = image_spec_small + image_spec_large
+
+            # Enable Group Mirroring and Verify
+            enable_group_mirroring_and_verify_state(
+                rbd_primary, **{"group-spec": group_spec}
+            )
+
+            # Wait for group mirroring to complete
+            wait_for_idle(rbd_primary, **{"group-spec": group_spec})
+            log.info("Data replay state is idle for all images in the group")
```

Comment:
also add "Data replay state is idle for all images in the group, Group mirroring is completed"